main.cpp

```diff
@@ -357,26 +397,26 @@ void InvoiceManager::Invoice::markAsPaid(Invoice& invoiceA)
     invoiceA.overdue = false;
 }
 
-void InvoiceManager::Invoice::duplicate(Invoice& invoiceA)
+void InvoiceManager::Invoice::duplicate(const Invoice& invoiceA)
 {
-    Invoice invoiceB = invoiceA;    // I would not really do this
-    ++invoiceB.invoiceNumber;
-    std::cout << "Duplicated invoice " << invoiceA.invoiceNumber << " as " << invoiceB.invoiceNumber << std::endl;
+    Invoice* invoiceB = new Invoice(invoiceA);
+    ++invoiceB->invoiceNumber;
+    std::cout << "Duplicated invoice " << invoiceA.invoiceNumber << " as " << invoiceB->invoiceNumber << std::endl;
```

Comment:
No.
What happens when you hit the closing curly brace of this function? 
you allocated this object on the heap.
Did you de-allocate it? 
- [ ] revise

Reply:
I spent some time trying to make this function valid, but ultimately it seems to be a nonsensical prototype that should be deleted at this point.